### ISSUE TITLE
unmet dependencies: libpcre2-posix2

### ISSUE THREAD

**Issue by leo-unglaub** [oerdnj/deb.sury.org#1744]
**Frequently asked questions**

- [*] I have read [Frequently Asked Questions](https://github.com/oerdnj/deb.sury.org/wiki/Frequently-Asked-Questions)

**Describe the bug**
The systems package manager bricks itself because a package from your PPA tries to overwrite a file already provided in a package from the main systems PPA. The only solution is to remove your PPA, fix the error and re-add it again. The problem wit that is, as soon as the package from base gets updated, your package manager bricks itself again.

You start out with the following error message: 
```
$ sudo apt upgrade
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
You might want to run 'apt --fix-broken install' to correct these.
The following packages have unmet dependencies:
 libpcre2-dev : Depends: libpcre2-posix2 (= 10.39-2+ubuntu21.10.1+deb.sury.org+1) but it is not installed
E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).
```

This would be easily fixable, so lets add the missing dependency `libpcre2-posix2`. But during that install dpkg errors out:
```
$ sudo apt install libpcre2-posix2

Unpacking libpcre2-posix2:amd64 (10.39-2+ubuntu21.10.1+deb.sury.org+1) ...
dpkg: error processing archive /var/cache/apt/archives/libpcre2-posix2_10.39-2+ubuntu21.10.1+deb.sury.org+1_amd64.deb (--un
pack):
 trying to overwrite '/usr/lib/x86_64-linux-gnu/libpcre2-posix.so.3', which is also in package libpcre2-posix3:amd64 10.37-
0ubuntu2
Errors were encountered while processing:
 /var/cache/apt/archives/libpcre2-posix2_10.39-2+ubuntu21.10.1+deb.sury.org+1_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

Both packages `libpcre2-posix2` and `libpcre2-posix3` contain `/usr/lib/x86_64-linux-gnu/libpcre2-posix.so.3`. That is a problem and i think this could easily be solved with a different naming. I guess this could be solved also by pinning a package to a speciffic version, but this would cause errors later down the path and it seams to me like a hacky solution.

What are your thoughts on this? Thanks for looking into it.

**Distribution (please complete the following information):**
 - OS: `Ubuntu`
 - uname: `Linux desktop 5.16.11-76051611-generic #202202230823~1646248261~21.10~2b22243 SMP PREEMPT Wed Mar 2 20: x86_64 x86_64 x86_64 GNU/Linux`
 - Architecture: `amd64`
 - Repository: `packages.sury.org`

**Package(s) (please complete the following information):**
```
$ sudo apt policy libpcre2-posix2
libpcre2-posix2:
  Installed: (none)
  Candidate: 10.39-2+ubuntu21.10.1+deb.sury.org+1
  Version table:
     10.39-2+ubuntu21.10.1+deb.sury.org+1 500
        500 http://ppa.launchpad.net/ondrej/php/ubuntu impish/main amd64 Packages
```
```
$ sudo apt policy libpcre2-posix3
libpcre2-posix3:
  Installed: 10.37-0ubuntu2
  Candidate: 10.37-0ubuntu2
  Version table:
 *** 10.37-0ubuntu2 500
        500 http://us.archive.ubuntu.com/ubuntu impish/main amd64 Packages
        500 http://de.archive.ubuntu.com/ubuntu impish/main amd64 Packages
        100 /var/lib/dpkg/status
```

**Comment by mxdpeep** [comment]
just a note...

[image: image.png]
you should move forward with Impish Indri fast... in my opinion, 20.04.4 is
the best option right now

*Filip Oščádal*
*https://mxd.cz <https://mxd.cz>*

**Comment by eguaj** [comment]
I also got this problem on Ubuntu 20.04 this morning (with all PHP versions from 5.6 to 7.4 installed in parallel).

I resolved it by uninstalling `libpcre2-posix2` manually (which, in my case, required uninstalling `libpcre2-dev`, `php7.3-dev` and `php7.4-dev`, and reinstalling `php7.x-dev`):

```
$ sudo apt-get purge libpcre2-posix2 libpcre2-dev php7.3-dev php7.4-dev
$ sudo apt-get install php7.3-dev php7.4-dev
```

**Comment by ghubz** [comment]
Running 18.04.6 LTS and I get the following:

```Preparing to unpack .../libpcre2-posix3_10.39-3+ubuntu18.04.1+deb.sury.org+1_amd64.deb ...
Unpacking libpcre2-posix3:amd64 (10.39-3+ubuntu18.04.1+deb.sury.org+1) ...
dpkg: error processing archive /var/cache/apt/archives/libpcre2-posix3_10.39-3+ubuntu18.04.1+deb.sury.org+1_amd64.deb (--unpack):
 trying to overwrite '/usr/lib/x86_64-linux-gnu/libpcre2-posix.so.3.0.1', which is also in package libpcre2-posix2:amd64 10.39-2+ubuntu18.04.1+deb.sury.org+1
Errors were encountered while processing:
 /var/cache/apt/archives/libpcre2-posix3_10.39-3+ubuntu18.04.1+deb.sury.org+1_amd64.deb```

**Comment by oerdnj** [comment]
Ah, thanks.  I forgot to add  conflict with the broken `libposix2-pcre2 >= 10.39` package, this should be fixed in: `libpcre2-posix3_10.39-3+ubuntu<ver>+deb.sury.org+2` that should be available any time soon.

**Comment by tersmitten** [comment]
Fixed for me on 18.04

**Comment by shakaran** [comment]
Having the same at Ubuntu 22.10 kinetic (cannot install php8.1-dev), but sadly since it is not a LTS, debsury don't offer support.

libpcre2-dev : Depends: libpcre2-8-0 (= 10.40-1ubuntu1) but 10.40-1+ubuntu22.04.1+deb.sury.org+1 is to be installed
